build failures with gcc 9.2.1: sigcontext define conflict; drgui override warning

dynamorio does not compile on ubuntu 19.10.  There are mutually incompatible versions of sigcontext.h that are co-included.  It compiles fine for me on ubuntu 18.04 and ubuntu 16.10

```
/mnt/robhenry/dynamorio/dynamorio/core/unix/include/sigcontext.h:20: error: "FP_XSTATE_MAGIC2_SIZE" redefined [-Werror]
   20 | #define FP_XSTATE_MAGIC2_SIZE sizeof(FP_XSTATE_MAGIC2)
      | 
from /mnt/robhenry/dynamorio/dynamorio/core/drlibc/drlibc.c:43:
/usr/include/x86_64-linux-gnu/bits/sigcontext.h:29: note: this is the location of the previous definition
   29 | #define FP_XSTATE_MAGIC2_SIZE sizeof (FP_XSTATE_MAGIC2)
      | 
```
gcc is gcc 9.2.1-9ubuntu2

1%] Built target generate_syslog
[  1%] Built target api_headers
[  1%] Building C object core/CMakeFiles/drlibc.dir/drlibc/drlibc.c.o
In file included from /mnt/robhenry/dynamorio/dynamorio/core/unix/os_public.h:46,
                 from /mnt/robhenry/dynamorio/dynamorio/core/unix/os_exports.h:47,
                 from /mnt/robhenry/dynamorio/dynamorio/core/drlibc/../globals.h:348,
                 from /mnt/robhenry/dynamorio/dynamorio/core/drlibc/drlibc.c:43:
/mnt/robhenry/dynamorio/dynamorio/core/unix/include/sigcontext.h:20: error: "FP_XSTATE_MAGIC2_SIZE" redefined [-Werror]
   20 | #define FP_XSTATE_MAGIC2_SIZE sizeof(FP_XSTATE_MAGIC2)
      |
In file included from /usr/include/signal.h:291,
                 from /mnt/robhenry/dynamorio/dynamorio/core/lib/globals_shared.h:125,
                 from /mnt/robhenry/dynamorio/dynamorio/core/drlibc/../globals.h:93,
                 from /mnt/robhenry/dynamorio/dynamorio/core/drlibc/drlibc.c:43:
/usr/include/x86_64-linux-gnu/bits/sigcontext.h:29: note: this is the location of the previous definition
   29 | #define FP_XSTATE_MAGIC2_SIZE sizeof (FP_XSTATE_MAGIC2)
      |
cc1: all warnings being treated as errors
make[2]: *** [core/CMakeFiles/drlibc.dir/build.make:83: core/CMakeFiles/drlibc.dir/drlibc/drlibc.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:212: core/CMakeFiles/drlibc.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

I have a dev machine with gcc 9.2.1 in FC31 which hits this sigcontext error, and a drgui warning:
```
In file included from dynamorio/ext/drgui/moc_drgui_main_window.cpp:9:
dynamorio/ext/drgui/../../../../src/dynamorio/ext/drgui/drgui_main_window.h:67:5: error: ‘virtual void drgui_main_window_t::closeEvent(QCloseEvent*)’ can be marked override [-Werror=suggest-override]
   67 |     closeEvent(QCloseEvent *event);
      |     ^~~~~~~~~~
```

I will fix these as they are in front of me and blocking me.  I do not plan to test Ubuntu 19.10 -- those using that should send pull requests to fix issues there.

--- a/core/unix/include/sigcontext.h
+++ b/core/unix/include/sigcontext.h
@@ -17,7 +17,9 @@
 
 #define FP_XSTATE_MAGIC1 0x46505853U
 #define FP_XSTATE_MAGIC2 0x46505845U
-#define FP_XSTATE_MAGIC2_SIZE sizeof(FP_XSTATE_MAGIC2)
+#ifndef FP_XSTATE_MAGIC2_SIZE
+#    define FP_XSTATE_MAGIC2_SIZE sizeof(FP_XSTATE_MAGIC2)
+#endif
 
 /*
  * bytes 464..511 in the current 512byte layout of fxsave/fxrstor frame

--- a/ext/drgui/drgui_main_window.h
+++ b/ext/drgui/drgui_main_window.h
@@ -64,7 +64,7 @@ public:
 
 protected:
     void
-    closeEvent(QCloseEvent *event);
+    closeEvent(QCloseEvent *event) override;
 
 private slots:
     void

